pkgs/by-name/hy/hyprviz/package.nix

```
@@ -0,0 +1,50 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  gtk4,
+  glib,
+  pango,
+  pkg-config,
+  wrapGAppsHook4,
+  hyprland,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hyprviz";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "timasoft";
+    repo = "hyprviz";
+    rev = "v${version}";
+    hash = "sha256-chZtqB74TMblNZy+aGeRsq8wleAyFdCRz7ooLTuf+i0=";
+  };
+
+  cargoHash = "sha256-f9A7qOFYmPdNwD109k0Q6vXXO8NJFtAmguvk6Mz3XBw=";
+
+  nativeBuildInputs = [
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    gtk4
+    glib
+    pango
+  ];
+
+  postInstall = ''
+    install -Dm644 hyprviz.desktop -t $out/share/applications
+  '';
+
+  meta = with lib; {
+    description = "GUI for configuring Hyprland";
+    homepage = "https://github.com/timasoft/hyprviz";
+    license = licenses.gpl2Only;
+    maintainers = with maintainers; [ timasoft ];
+    mainProgram = "hyprviz";
+    platforms = lib.platforms.linux;
```

you don't seem to use `hyprland` anywhere in your code, but here's a usage for it
```suggestion
    platforms = hyprland.meta.platforms;
```